docs/architecture/adr-0028-tracing.md

```diff
@@ -0,0 +1,96 @@
+# Context
+
+# The Contenders
+
+I considered the following tracing libraries, presented in the order in which I
+tried them out. Each of them is seemingly maintained by one person, so there's
+considerable risk that development will stop someday at random on each of them.
+
+### `natchez`
+
+Natchez promises nice integration with things that we already use from the
+TypeLevel ecosystem, or at least it would, if I could figure out how to provide
+a `ConcurrentEffect` for `Kleisli[F, Span[F], ?]`. The reasons that's hard are:
+
+- I'm having trouble figuring out how to construct a `new ConcurrentEffect[F]`,
+not just for this particular evidence, but in general. When I try to implement
+a `runCancelable`, I get a warning about `runCancelable` being unused,
+despite that appearing to be the method definition that [I'm supposed to care
+about](https://github.com/typelevel/cats-effect/blob/master/core/shared/src/main/scala/cats/effect/ConcurrentEffect.scala#L42-L61).
+- The most obvious workaround is suggested by
+the `EitherT` concurrent (or `WriterT`, or take your
+pick), but the trait I'd need (`KleisliConcurrent`) is [private to
+`cats-effect`](https://github.com/typelevel/cats-effect/blob/master/core/shared/src/main/scala/cats/effect/Concurrent.scala#L749).
+- Moreover, the [`natchez`
+example](https://github.com/tpolecat/natchez/blob/master/modules/examples/src/main/scala/Example.scala)
+has no idea about being a webservice -- it only knows how to ask a database for
+some information. The extra typeclass requirements from `BlazeServerBuilder`
+are a bit challenging to resolve.
+
+Rob Norris currently recommends not to use `natchez` in
+anything you need to work, and I think we should take his
+[advice](https://github.com/tpolecat/natchez/blame/efcefbf28c88ab977206df0f394d4141b3c5b2ca/README.md#L60).
+
+The (broken) implementation I wound up with is on
+[`feature/js/natchez`](https://github.com/jisantuc/tracing-demos/tree/feature/js/natchez).
```

Comment:
There is a [discussion](https://github.com/tpolecat/natchez/issues/5) on what an `http4s` module for `natchez` would look like -- I'm not sure if it helps in this case, though since nothing is implemented yet it might be a useful signal to hold off on any integration.

Reply:
I think it's at least worth mentioning that the discussion exists in the ADR. I've read Rob Norris's comment there like six times now but haven't had coffee yet so it hasn't stuck with me (really struggled with whether to @ him and decided probably not).